src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const remProp of action.keysToRemove) {
+          if (state[remProp]) {
+            delete state[remProp];
+          }
+        };
+        break;
+
+      case 'clear':
+        for (const prop in state) {
+          delete state[prop];
+        }
+        break;
+    }
```

Comment:
it is a good practice when you add a default behavior for switch statement. Yet, if you don't have such one - just `throw an Error` with text like `"unexpected action"`